### PR TITLE
graphql.execute can accept a builder

### PR DESCRIPTION
**Pull request by bbakerman**
Like the other parts of the code that use a builder, it now takes one as input plus used the builder function pattern

### CODE REVIEW

File: src/main/java/graphql/GraphQL.java [hunk 1]
```diff
@@ -178,11 +179,11 @@ public GraphQL build() {
...
 
     /**
+     * Executes the specified graphql query/mutation/subscription
+     *
      * @param query the query/mutation/subscription
      * @return result including errors
-     * @deprecated Use {@link #execute(ExecutionInput)}
      */
-    @Deprecated
     public ExecutionResult execute(String query) {
```
Comment (bbakerman) [inline src/main/java/graphql/GraphQL.java]: This is one contentious aspect of this PR.

The most basic way to execute a query (just the string) was deprecated.

50 of our tests are now invalid.  It tells me that we need a simple bit of sugar to allow an easy way to get going and then have the ExecutionInput as the way to do it in a more complex way.

File: src/test/groovy/graphql/GraphQLTest.groovy [hunk 2]
```diff
@@ -334,4 +340,30 @@ class GraphQLTest extends Specification {
...
+    def "execution input using builder function"() {
+        given:
+        GraphQLSchema schema = simpleSchema()
+
+        when:
+
+        def builderFunction = { it.query('{hello}')} as UnaryOperator<ExecutionInput.Builder>
+        def result = GraphQL.newGraphQL(schema).build().execute(builderFunction).data
+
+        then:
```
Comment (bbakerman) [inline src/test/groovy/graphql/GraphQLTest.groovy]: groovy and java closures are not so sweet :(

File: src/main/java/graphql/GraphQL.java (hunk 1 above)
Comment (andimarek) [inline src/main/java/graphql/GraphQL.java]: yes, I totally agree: `execute(String)` should not be deprecated

File: src/test/groovy/graphql/GraphQLTest.groovy (hunk 2 above)
Comment (andimarek) [inline src/test/groovy/graphql/GraphQLTest.groovy]: yes :(

**Pull request merged**